lib/ansible/modules/cloud/vmware/vmware_guest_tools.py

```diff
@@ -0,0 +1,246 @@
+    - python >= 2.6
+    - PyVmomi
+options:
+   name:
+        description:
+            - Name of the VM for which to wait until the tools become available
+            - This is required if uuid is not supplied.
+   name_match:
+        description:
+            - If multiple VMs matching the name, use the first or last found
+        default: 'first'
+        choices: ['first', 'last']
+   folder:
+        description:
+            - Destination folder, absolute or relative path to find an existing guest.
+            - This is required if name is supplied.
+            - The folder should include the datacenter. ESX's datacenter is ha-datacenter
+            - 'Examples:'
+            - '   folder: /ha-datacenter/vm'
+            - '   folder: ha-datacenter/vm'
+            - '   folder: /datacenter1/vm'
+            - '   folder: datacenter1/vm'
+            - '   folder: /datacenter1/vm/folder1'
+            - '   folder: datacenter1/vm/folder1'
+            - '   folder: /folder1/datacenter1/vm'
+            - '   folder: folder1/datacenter1/vm'
+            - '   folder: /folder1/datacenter1/vm/folder2'
+        default: /vm
+   uuid:
+        description:
+            - UUID of the VM  for which to wait until the tools become available, if known. This is VMware's unique identifier.
+            - This is required if name is not supplied.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Wait for VMware tools to become available by UUID
+  vmware_guest_tools_wait:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    uuid: 421e4592-c069-924d-ce20-7e7533fab926
+  delegate_to: localhost
+  register: facts
+
+- name: Wait for VMware tools to become available by name
+  vmware_guest_tools_wait:
+    hostname: 192.168.1.209
+    username: administrator@vsphere.local
+    password: vmware
+    validate_certs: no
+    name: test-vm
+    folder: /datacenter1/vm
+  delegate_to: localhost
+  register: facts
+'''
+
+RETURN = """
+instance:
+    description: metadata about the virtual machine
+    returned: always
+    type: dict
+    sample: None
+"""
+
+import time
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import connect_to_api, gather_vm_facts, vmware_argument_spec, find_vm_by_id
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+HAS_PYVMOMI = False
+try:
+    import pyVmomi
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    pass
+
+
+class PyVmomiHelper(object):
+    def __init__(self, module):
+        if not HAS_PYVMOMI:
+            module.fail_json(msg='pyvmomi module required')
+
+        self.module = module
+        self.params = module.params
+        self.content = connect_to_api(self.module)
+
+    def getvm(self, name=None, uuid=None, folder=None):
```

Comment:
Can you use refactored [getvm](https://github.com/ansible/ansible/pull/27188/files#diff-d5ca044435f1fc628995b038d2e238f7R173) ?